### PR TITLE
created structure, sun module, guard, localStorage, cookies services,…

### PR DESCRIPTION
**Pull request by DmitrySkyrat**
… pagination, breadcrumbs

### CODE REVIEW

File: src/app/modules/shared/components/feedback/feedback.component.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    });
+
+  confirmValidator = (control: FormControl): { [s: string]: boolean } => {
+    if (!control.value) {
+      return { error: true, required: true };
+    } else if (control.value !== this.validateForm.controls.password.value) {
+      return { confirm: true, error: true };
+    }
+    return {};
+  };
```
Comment (Gorgchap) [inline src/app/modules/shared/components/feedback/feedback.component.ts:72]: Much better to keep validators in a separate file (validators.ts). There is a frequent use case such as reusage of validators in different forms.

File: src/app/modules/sun/components/sun/sun.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  constructor(private spaceHttpService: SpaceHttpService) {}
+
+  ngOnInit(): void {
+    this.items$ = combineLatest(
+      [this.spaceHttpService.getPictures('sun'), this.pageNumber$]
+    ).pipe(
+      map(([items, page]) => {
+        this.itemsAmount = items.length;
+        return [...items.slice((page - 1) * 10, (page - 1) * 10 + 10)]
+      })
```
Comment (Gorgchap) [inline src/app/modules/sun/components/sun/sun.component.ts:24]: Array.prototype.slice is a pure function (generates a new array), so there is no necessity to use spread.

File: src/app/modules/shared/components/feedback/feedback.component.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      email: ['', [Validators.email, Validators.required]],
+      password: ['', [Validators.required]],
+      confirm: ['', [this.confirmValidator]],
+      comment: ['', [Validators.required]]
+    });
+  }
+
+  showModal(): void {
+    this.isVisible = true;
+  }
```
Comment (Gorgchap) [inline src/app/modules/shared/components/feedback/feedback.component.ts:27]: I guess that Input() decorator is much more approriate (there is no complex logic in child component that should be implemented during parent's working process).

**Review by Gorgchap — CHANGES_REQUESTED** [review]

File: src/app/modules/sun/components/sun/sun.component.ts (hunk 2 above)
Comment (DmitrySkyrat) [reply src/app/modules/sun/components/sun/sun.component.ts:24]: done

File: src/app/modules/shared/components/feedback/feedback.component.ts (hunk 1 above)
Comment (DmitrySkyrat) [reply src/app/modules/shared/components/feedback/feedback.component.ts:72]: done

File: src/app/modules/shared/components/feedback/feedback.component.ts (hunk 3 above)
Comment (DmitrySkyrat) [reply src/app/modules/shared/components/feedback/feedback.component.ts:27]: I didn't catch this point)

**Comment by DmitrySkyrat** [comment]
fixed some issues

**Pull request merged**